### ISSUE TITLE
stdc_count_zeros(-1) gives -32

### ISSUE THREAD

**Issue by fragrans** [llvm/llvm-project#111344]
Ubuntu clang version 18.1.3 (1ubuntu1)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

**Comment by fragrans** [comment]
Ubuntu clang version 18.1.3 (1ubuntu1)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

**Comment by dtcxzyw** [comment]
@fragrans Can you provide godbolt links?

**Comment by fragrans** [comment]
Yes, but, it can't compile, because the its lack the stdbit.h for c23. 

https://godbolt.org/z/GbW8W5Wo6

**Comment by erichkeane** [comment]
Where are you seeing the definition of `stdc_count_zeros`?  The only implementation of this I see at the moment is llvm-libc, but this function is only defined (by standard) for unsigned types.  So perhaps this is UB? 

We start with a _Generic, which is:
```
#define stdc_count_ones(x)                                                     \
   _Generic((x),                                                                \
       unsigned char: stdc_count_ones_uc,                                       \
       unsigned short: stdc_count_ones_us,                                      \
       unsigned: stdc_count_ones_ui,                                            \
       unsigned long: stdc_count_ones_ul,                                       \
       unsigned long long: stdc_count_ones_ull)(x)
```

I see that we eventually just call:
`return cpp::popcount<T>(static_cast<T>(~value));`

as the implementation of this.  

So I suspect there is some sort of conversion here happening along the way (what does `_Generic` do if no type is specified? ) that is resulting in -1 being 0?

**Comment by AaronBallman** [comment]
Closing the issue as invalid due to the lack of a response or reproducer; if the issue persists, please reopen with more details.
